shard/committee/assignment.go

```diff
@@ -123,17 +125,47 @@ func eposStakedCommittee(
 		Int("staked-candidates", len(candidates)).
 		Msg("preparing epos staked committee")
 
+	shardCount := int(s.NumShards())
+	shardState := &shard.State{}
+	shardState.Shards = make([]shard.Committee, shardCount)
+	hAccounts := s.HmyAccounts()
+	shardHarmonyNodes := s.NumHarmonyOperatedNodesPerShard()
+
+	for i := 0; i < shardCount; i++ {
+		shardState.Shards[i] = shard.Committee{uint32(i), shard.SlotList{}}
+		for j := 0; j < shardHarmonyNodes; j++ {
+			index := i + j*shardCount
+			pub := &bls.PublicKey{}
+			if err := pub.DeserializeHexStr(hAccounts[index].BlsPublicKey); err != nil {
```

Comment:
is it possible index is out of bound? do you need to check it?

Reply:
no, this is original same code as been running from ~july 2019